Add instructions for setting up a restricted-by-default local cluster for Rancher (#423)
rancher/rancher-docs: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md

* `rancher-restricted`:  This policy follows current pod-hardening best practices.

If you are a Rancher administrator or have restricted administrator privileges, you can customize restrictions and permissions by creating additional PSAC templates, or by editing existing templates.

:::caution
If you edit an existing PSAC template while it is still in use, it *will* affect all clusters that have been assigned that template.
:::

If you want to allow users other than the Rancher administrator to manage PSAC templates, you can bind the user to a role that grants the user all verbs ("*") on `management.cattle.io/podsecurityadmissionconfigurationtemplates`.

:::warning
Any user that is bound to the above permission will be able to change the restriction levels on all managed clusters which use this template, including ones they have no permissions on.
:::

### Rancher On PSA-restricted Clusters

When you run Rancher on a Kubernetes cluster that enforces the `rancher-restricted` PSS policy by default, you'll need to exempt the following namespaces. This prevents the policy from blocking Rancher system pods:

Here is the list of the namespaces that require an exemption:

> * `kube-system`

Reviewer: For consistency with our other content, could you please use `-` to start the items in this list. `-` is preferable to `*` for readability since `*` is used to italicize content.

title: Pod Security Standards
---

[Pod Security Standards (PSS)](https://kubernetes.io/docs/concepts/security/pod-security-standards/) define security restrictions for a broad set of workloads.  They replace [Pod Security Policies](https://kubernetes.io/docs/concepts/security/pod-security-policy/), which are now deprecated.

## Upgrade to Pod Security Standards

Make sure that all Pod Security Policies have been migrated, which should result in equivalent security controls still being enforced (e.g. Pod Security Admission, Gatekeeper, Kyverno) and all PodSecurityPolicy objects removed from the cluster.

:::info important

> The new policy enforcement must be added before the PodSecurityPolicy objects are removed, or risk creating a window of opportunity for privilege escalation within the cluster.

Reviewer 2: or **you** risk creating...

title: Pod Security Standards
---

> [Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/) define
> security restrictions for a broad set of workloads. 
> They replace [Pod Security Policies](https://kubernetes.io/docs/concepts/security/pod-security-policy/),
> which are now deprecated.

Reviewer 3: ```suggestion
[Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/) define security restrictions for a broad set of workloads.  They replace [Pod Security Policies](https://kubernetes.io/docs/concepts/security/pod-security-policy/), which are now deprecated.
```

Please update other instances where there are line breaks in the middle of sentences. Unfortunately, the line breaks get rendered to the published page.

Author: Ah, ok.  Sorry about that.  I usually default to keeping lines short.

	* That updating a template will affect live clusters
	* Who can edit. Including a template management role workaround with it's own warning/caveat.

> * How to upgrade

Reviewer 3: Lines 15 to 21: This should be filled out or removed before we publish.

> 	* Ensure that a replacement is in place before you upgrade and PSPs are off

Reviewer 3: ```suggestion
	:::important
	
Make sure that you've set up a replacement and that Pod Security Policies (PSPs) are off before you try to upgrade.
	
	:::
```

I rephrased but we should specify what is being replaced (a replacement x?)

Reviewer 4: I would let @jiaqiluo make the wider changes about how to upgrade and what things to do. 

This statement is specifically about security, so @pjbgf would you care to add a suggestion for a warning. 

Also this whole section is a "todo" i made for big things we need to document, so it all needs to be replaced soon.

Reviewer 5: Feel free to embellish it:

Make sure that all [Pod Security Policies were migrated], which should result in equivalent security controls still being enforced (e.g. [Pod Security Admission], [Gatekeeper], [Kyverno]) and all `PodSecurityPolicy` objects removed from the cluster.
The new policy enforcement must be added **before** the `PodSecurityPolicy` objects are removed, or risk creating a window of opportunity for privilege escalation within the cluster.

[Pod Security Policies were migrated]: https://kubernetes.io/docs/tasks/configure-pod-container/migrate-from-psp/
[Pod Security Admission]: https://kubernetes.io/docs/concepts/security/pod-security-admission/
[Gatekeeper]: https://github.com/open-policy-agent/gatekeeper
[Kyverno]: https://github.com/kyverno/kyverno/

> ### Running Rancher on a local cluster that is PSA restricted by default

Reviewer 3: If this is the only header on the page, this should be the title. Or, if we add more information before the header, this section should be made into a second-level header. It seems like some content went missing and an early version of the draft got marked ready for review.

	* That updating a template will affect live clusters
	* Who can edit. Including a template management role workaround with it's own warning/caveat.
* How to upgrade
	* Ensure that a replacement is in place before you upgrade and PSPs are off

### Rancher On PSA-restricted Clusters

> When you run Rancher on a Kubernetes cluster that enforces the `rancher-restricted` PSS policy by default,

Reviewer 4: ```suggestion
When you run Rancher on a Kubernetes cluster that enforces a restrictive PSS policy by default, such as the `rancher-restricted` PSS policy,
```
